circuit/list.go

```diff
@@ -0,0 +1,77 @@
+package circuit
+
+// a simple list to keep breakers sorted by access order.
+// It can return the first consecutive items that match a
+// condition (in our case, the ones that were inactive for a
+// while).
+type list struct {
```

Comment:
now that thinking about it.. maybe there is a way to bind the breakers to each route. Will play with that as a separate effort.